ReactSkia/sdk/Alert.cpp

```diff
@@ -0,0 +1,230 @@
+                            alertRef.alertMessage.length(),
+                            SkTextEncoding::kUTF8,
+                            drawStartPointX,
+                            drawStartPointY,
+                            font_,
+                            paint_);
+
+    msgAndTitleDirtyRect.join(SkIRect::MakeXYWH(drawStartPointX,drawStartPointY-textFontSize_,
+                               messageWidth,textFontSize_+lineSpacing_));
+    RNS_LOG_DEBUG(" drawAlert Msg :: "<<alertRef.alertMessage.c_str());
+  }
+
+  dirtyRect.push_back(msgAndTitleDirtyRect);
+}
+
+inline void Alert::removeAlertFromAlertList(unsigned int msgIndex) {
+
+  if(alertInfoList_.empty() || (msgIndex > alertInfoList_.size())) {
+    return;
+  }
+
+  if(alertInfoList_.size() == idOfMessageOnDisplay_) {
+    // Current Alert Message is the Latest Message
+    alertInfoList_.pop_back();
+  } else {
+    //New Alert Message Pending to be displayed
+    std::list<alertInfo>::iterator it = alertInfoList_.begin();
+    std::advance(it, msgIndex);
+    alertInfoList_.erase(it);
+  }
+}
+
+void Alert::windowReadyToDrawCB() {
+
+  setWindowTittle("Alert Window");
+
+//Get Darw Parameters for screen Size
+  textFontSize_=TEXT_FONT_SIZE*(alertWindowSize_.width()/DEFAULT_SCREEN_WIDTH);
+  lineSpacing_=LINE_SPACING*(alertWindowSize_.height()/DEFAULT_SCREEN_HEIGHT);
+  font_.setSize(textFontSize_);
+  paint_.setColor(DARK_THEME_FONT_COLOR);
+  paint_.setAntiAlias(true);
+
+  alertWindowState_ = ALERT_WINDOW_ACTIVE;
+
+  triggerRenderRequest(ALERT_BACKGROUND,true);//Batch this render call along with Below one
+  triggerRenderRequest(ALERT_TITLE_AND_MESSAGE);
+}
+
+void Alert::createAlertWindow() {
+  alertWindowSize_ = RnsShell::Window::getMainWindowSize();
+  
+  std::function<void()> createWindowCB = std::bind(&Alert::windowReadyToDrawCB, this);
+  createWindow(alertWindowSize_, createWindowCB);
+
+  if (subWindowKeyEventId_ == -1) {
+    std::function<void(rnsKey, rnsKeyAction,RnsShell::Window*)> handler = std::bind( &Alert::onHWKeyHandler,this,
+                                std::placeholders::_1,  // KeyValue
+                                std::placeholders::_2,  // eventKeyAction
+                                std::placeholders::_3);  // window onject
+    subWindowKeyEventId_ = NotificationCenter::subWindowCenter().addListener("onHWKeyEvent", handler);
+  }
+}
+
+void Alert::onHWKeyHandler(rnsKey keyValue,rnsKeyAction eventKeyAction,RnsShell::Window *window) {
+  RNS_LOG_DEBUG("KEY RECEIVED : " << RNSKeyMap[keyValue]);
+  if ((getWindow() != window) || (eventKeyAction != RNS_KEY_Press) || (RNS_KEY_Select != keyValue) ) {
+    return;
+  }
+  if(idOfMessageOnDisplay_ == -1) {
+    return;// No Valid Index, nothing to remove from display
+  }
+
+  msgPendingToBeChanged_=true;
+  unsigned int msgIndex=idOfMessageOnDisplay_;
+
+  std::lock_guard<std::mutex> lock(alertListAccessCtrlMutex_);
+
+  removeAlertFromAlertList(msgIndex);
+  msgPendingToBeChanged_=false;
+
```

Comment:
after remove should we decrement the idOfMessageOnDisplay_ ?
else we will be checking with old value of idOfMessageOnDisplay_ here at line 167 right

Reply:
idOfMessageOnDisplay_  will be updated from the below line "triggerRenderRequest" while drawing new alert message on the screen. So before this API finishes ID will be updated.